src/ext/java/org/opentripplanner/ext/vehicletostopheuristics/VehicleToStopSkipEdgeStrategy.java

```diff
@@ -70,12 +72,13 @@ public boolean shouldSkipEdge(State current, Edge edge) {
         current.getVertex() instanceof TransitStopVertex stopVertex &&
         !stopsCounted.contains(stopVertex.getStop().getId())
       ) {
+        // TODO: 2022-12-05 filters: check performance on that and verify that this is right. Previously we were filtering just on modes
         var stop = stopVertex.getStop();
         var score = getRoutesForStop
           .apply(stop)
           .stream()
+          .filter(route -> filters.stream().anyMatch(f -> f.matchRoute(route)))
```

Comment:
I'm a bit worried about creating new streams for every stop, which has shown to be bad for performance. If it proves to be slow, I can fix that myself however.

Reply:
I converted this code to classic for statement